src/util/parse_options.cpp

```diff
@@ -28,11 +28,12 @@ parse_options_baset::parse_options_baset(
   const std::string &_optstring,
   int argc,
   const char **argv,
-  message_handlert &mh)
-  : log(mh)
+  const std::string &program)
 {
   std::string optstring=std::string("?h(help)")+_optstring;
   parse_result=cmdline.parse(argc, argv, optstring.c_str());
+  message_handler = util_make_unique<ui_message_handlert>(cmdline, program);
+  log = messaget{*message_handler};
```

Comment:
I really think that `message_handler` should be a full member (see also comment below) and the constructor should do `: message_handler(cmdline, program), log(message_handler)`

Reply:
I've done the change now. The reason I didn't do it originally was that it required a bit of reordering because the command line needs to be parsed before the object is constructed, but the change is actually not too bad.